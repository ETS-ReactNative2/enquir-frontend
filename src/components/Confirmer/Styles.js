import { StyleSheet } from 'react-native';

const containers = StyleSheet.create({
  modalBackground: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-around',
    backgroundColor: '#00000040'
  },
  confirmerWrapper: {
    backgroundColor: '#FFFFFF',
    height: 140,
    width: 260,
    borderWidth: .5,
    borderColor: 'grey',
    borderRadius: 8,
    display: 'flex',
    justifyContent: 'center',
  },
  buttons: {
    position: 'absolute',
    bottom: 0,
    alignSelf: 'center',
    flexDirection: 'row',
  },
});

const text = StyleSheet.create({
  confirmation: {
    textAlign: 'center',
    fontSize: 17,
    fontFamily: 'Roboto-Medium',
    marginHorizontal: 12,
    marginBottom: 20,
  },
  cancel: {
    marginTop: 5,
    color: 'grey',
    textAlign: 'center',
    fontSize: 18,
    fontFamily: 'Roboto-Medium',
  },
  discard: {
    marginTop: 5,
    color: 'red',
    textAlign: 'center',
    fontSize: 18,
    fontFamily: 'Roboto-Medium',
  },
  save: {
    marginTop: 5,
    color: 'blue',
    textAlign: 'center',
    fontSize: 18,
    fontFamily: 'Roboto-Medium',
  },
});

const buttons = StyleSheet.create({
  cancelButton: {
    flex: 1,
    borderTopWidth: .75,
    borderColor: 'grey',
    borderBottomLeftRadius: 8,
  },
  discardButton: {
    flex: 1,
    borderTopWidth: .75,
    borderLeftWidth: .65,
    borderRightWidth: .65,
    borderColor: 'grey',
    height: 40,
  },
  saveButton: {
    flex: 1,
    borderTopWidth: .75,
    borderColor: 'grey',
    borderBottomRightRadius: 8,
  },
});

export { containers, text, buttons };