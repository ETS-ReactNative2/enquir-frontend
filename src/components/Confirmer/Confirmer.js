import React from 'react';
import {
  View,
  Modal,
  Text,
  TouchableOpacity,
} from 'react-native';
import { containers, text, buttons } from './Styles';

const Confirmer = props => {
  const {
    confirming,
    confirmation,
  } = props;

  return (
    <Modal
      transparent={true}
      animationType={'none'}
      visible={confirming} >
      <View style={containers.modalBackground}>
        <View style={containers.confirmerWrapper}>
          <Text style={text.confirmation}>You have unsaved changes.</Text>
          <View style={containers.buttons}>
            <TouchableOpacity
              onPress={() => confirmation(null)}
              style={buttons.cancelButton} >
              <Text style={text.cancel}>Cancel</Text>
            </TouchableOpacity>
            <TouchableOpacity
              onPress={() => confirmation("discard")}
              style={buttons.discardButton} >
                <Text style={text.discard}>Discard</Text>
            </TouchableOpacity>
            <TouchableOpacity
              onPress={() => confirmation("save")}
              style={buttons.saveButton} >
                <Text style={text.save}>Save</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </Modal>
  )
};

export default Confirmer;