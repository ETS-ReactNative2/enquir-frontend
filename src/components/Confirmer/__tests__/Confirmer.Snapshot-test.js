import React from 'react';
import renderer from 'react-test-renderer';

import Confirmer from '../Confirmer';

test('renders a Confirmer using Snapshots', () => {
    expect(renderer.create(
        <Confirmer
            confirming={false}
            confirmation={jest.fn()}
        />
    )).toMatchSnapshot();
});