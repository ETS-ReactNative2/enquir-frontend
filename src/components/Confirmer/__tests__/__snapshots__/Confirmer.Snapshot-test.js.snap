// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Confirmer using Snapshots 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "grey",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "width": 260,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Roboto-Medium",
            "fontSize": 17,
            "marginBottom": 20,
            "marginHorizontal": 12,
            "textAlign": "center",
          }
        }
      >
        You have unsaved changes.
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "bottom": 0,
            "flexDirection": "row",
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderBottomLeftRadius": 8,
              "borderColor": "grey",
              "borderTopWidth": 0.75,
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontFamily": "Roboto-Medium",
                "fontSize": 18,
                "marginTop": 5,
                "textAlign": "center",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "grey",
              "borderLeftWidth": 0.65,
              "borderRightWidth": 0.65,
              "borderTopWidth": 0.75,
              "flex": 1,
              "height": 40,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontFamily": "Roboto-Medium",
                "fontSize": 18,
                "marginTop": 5,
                "textAlign": "center",
              }
            }
          >
            Discard
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderBottomRightRadius": 8,
              "borderColor": "grey",
              "borderTopWidth": 0.75,
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "fontFamily": "Roboto-Medium",
                "fontSize": 18,
                "marginTop": 5,
                "textAlign": "center",
              }
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
